spec/abstractops.html

```diff
@@ -45,59 +45,35 @@ <h1>GetOptionsObject ( _options_ )</h1>
     </emu-alg>
   </emu-clause>
 
-  <!-- Copied from ECMA-402 GetOption -->
+  <!-- Based on ECMA-402 GetOption -->
   <emu-clause id="sec-getoption" aoid="GetOption">
-    <h1>GetOption ( _options_, _property_, _type_, _values_, _fallback_ )</h1>
+    <h1>GetOption ( _options_, _property_, _types_, _values_, _fallback_ )</h1>
 
     <p>
-      The abstract operation GetOption extracts the value of the property named _property_ from the provided _options_ object, converts it to the required _type_, checks whether it is one of a List of allowed _values_, and fills in a _fallback_ value if necessary. If _values_ is *undefined*, there is no fixed set of values and any is permitted.
+      The abstract operation GetOption extracts the value of the property named _property_ from the provided _options_ object, checks whether its type is one of a List of allowed _types_ and converts it to the last one if not, checks whether it is one of a List of allowed _values_, and fills in a _fallback_ value if necessary. If _values_ is *undefined*, there is no fixed set of values and any is permitted.
     </p>
 
     <emu-alg>
       1. Assert: Type(_options_) is Object.
       1. Let _value_ be ? Get(_options_, _property_).
       1. If _value_ is *undefined*, return _fallback_.
-      1. Assert: _type_ is *"boolean"* or *"string"*.
-      1. If _type_ is *"boolean"*, then
-        1. Let _value_ be ! ToBoolean(_value_).
-      1. If _type_ is *"string"*, then
-        1. Let _value_ be ? ToString(_value_).
+      1. If _types_ contains Type(_value_), then
+        1. Let _type_ be Type(_value_).
+      1. Else,
+        1. Let _type_ be the last element of _types_.
+      1. Assert: _type_ is Boolean, String, or Number.
```

Comment:
this kind of obscurely captures "_types_ must only contain Boolean, String, or Number" - perhaps that assertion at the top of the AO would be clearer than waiting til now? (perhaps both?)